scripts/scripts.js

```diff
@@ -44,6 +47,45 @@ async function loadFonts() {
   }
 }
 
+/**
+ * Builds an article header and prepends to main in a new section.
+ * @param main
+ */
+function buildArticleHeader(main) {
+  if (main.querySelector('.article-header')) {
+    // already got an article header
+    return;
+  }
+
+  const div = document.createElement('div');
+  const h1 = main.querySelector('h1');
+
+  const author = getMetadata('author');
+  const authorURL = getMetadata('author-url') || `/authors/${toClassName(author)}`;
+  const publicationDate = formatDate(getMetadata('publication-date'));
+
+  const articleHeaderBlock = buildBlock('article-header', [
+    [h1],
+    [`<p><a href="${authorURL}">${author}</a></p>
+      <p>${publicationDate}</p>`],
+  ]);
+  div.append(articleHeaderBlock);
+  main.prepend(div);
```

Comment:
If I may make a suggestion here, I wrote a small [dom helper](https://github.com/hlxsites/servicenow/blob/main/scripts/dom-helpers.js) script in an attempt to make the JS look as close as possible to the resulting dom. Using it would look like:

```javascript
main.prepend(
  div( //section
    buildBlock('article-header', [
      [
        main.querySelector('h1') 
      ],
      [
        p(a{ href: authorURL}, author}),
        p(publicationDate),
      ]
    ]);
  ),
)
```

Reply:
nice idea.